src/Core/Managed/Shared/Channel/Transmission.cs

```diff
@@ -131,14 +131,19 @@ public string Id
         /// Executes the request that the current transmission represents.
         /// </summary>
         /// <returns>The task to await.</returns>
+#if NET40
+        public virtual Task<HttpWebResponseWrapper> SendAsync()
+#else
         public virtual async Task<HttpWebResponseWrapper> SendAsync()
+#endif
         {
             if (Interlocked.CompareExchange(ref this.isSending, 1, 0) != 0)
             {
                 throw new InvalidOperationException("SendAsync is already in progress.");
             }
-
+#if !NET40
```

Comment:
due to the complex #if , this method is hard to read and understand. I would prefer to see a global (method-level) #if and then two clean implementations for 4.0 and not 4.0.